.github/workflows/ci.yml

```diff
@@ -0,0 +1,74 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: ⬇️ Checkout
+        uses: actions/checkout@v2
+
+      - name: ⎔ Setup node ${{ matrix.node }}
+        uses: actions/setup-node@v2
+        with:
+          cache: npm
+
+      - name: 📥 Download deps
+        run: npm ci
+
+      - name: 🧪 Run lint
+        run: npm run lint
+
+  build:
+    runs-on: ${{ matrix.os }}
+    needs: lint
+    strategy:
+      fail-fast: false
+      matrix:
+        node:
+          - 12
+          - 14
+          - 16
+          - 17
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        browser:
+          - FirefoxHeadless
```

Comment:
Edge and Safari runners seems to be abandoned and a little flaky, so didn't add them in, but can be added on platform specific `includes:`